frontend/mr-admin-flate/src/components/utbetaling/OpprettDelutbetalingRow.tsx

```diff
@@ -0,0 +1,135 @@
+  onDelutbetalingChange,
+}: Props) {
+  const [endretBelop, setEndretBelop] = useState<number>(belop);
+  const [endretFrigjorTilsagn, setEndretFrigjorTilsagn] = useState<boolean>(frigjorTilsagn);
+  const [openRow, setOpenRow] = useState<boolean>(frigjorTilsagn);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const godkjentTilsagn = tilsagn.status === TilsagnStatus.GODKJENT;
+
+  function handleOnChange(belop: number, frigjorTilsagn: boolean) {
+    onDelutbetalingChange({
+      id: id,
+      tilsagnId: tilsagn.id,
+      belop: belop,
+      frigjorTilsagn: frigjorTilsagn,
+    });
+  }
+  const cellClass = error && "align-top";
+  return (
+    <Table.ExpandableRow
+      defaultOpen={opprettelse && true}
+      onOpenChange={() => setOpenRow(!openRow)}
+      open={openRow}
+      expansionDisabled={!frigjorTilsagn}
+      key={tilsagn.id}
+      content={
+        <>
+          {opprettelse && (
+            <AvvistAlert
+              header="Utbetaling returnert"
+              navIdent={opprettelse?.besluttetAv}
+              aarsaker={opprettelse?.aarsaker}
+              forklaring={opprettelse?.forklaring}
+              tidspunkt={opprettelse?.besluttetTidspunkt}
+            />
+          )}
+          {frigjorTilsagn && (
+            <Alert variant="warning">
+              Når denne utbetalingen godkjennes vil det ikke lengre være mulig å sende inn nye
+              utbetalinger fra dette tilsagnet
+            </Alert>
+          )}
+        </>
+      }
+    >
+      <Table.DataCell className={cellClass}>{formaterDato(tilsagn.periodeStart)}</Table.DataCell>
+      <Table.DataCell className={cellClass}>{formaterDato(tilsagn.periodeSlutt)}</Table.DataCell>
+      <Table.DataCell className={cellClass}>{tilsagnTypeToString(tilsagn.type)}</Table.DataCell>
+      <Table.DataCell className={cellClass}>{tilsagn.kostnadssted.navn}</Table.DataCell>
+      <Table.DataCell className={cellClass}>
+        {formaterNOK(tilsagn.beregning.output.belop)}
+      </Table.DataCell>
+      <Table.DataCell className={cellClass}>
+        <Checkbox
+          hideLabel
+          readOnly={!kanRedigere}
+          checked={endretFrigjorTilsagn}
+          onChange={(e) => {
+            setEndretFrigjorTilsagn(e.target.checked);
+            handleOnChange(belop, e.target.checked);
+          }}
+        >
+          Gjør opp tilsagn
+        </Checkbox>
+      </Table.DataCell>
+      <Table.DataCell className={error && "align-top"}>
+        {godkjentTilsagn && kanRedigere ? (
+          <TextField
+            size="small"
+            error={error}
+            label="Utbetales"
+            hideLabel
+            inputMode="numeric"
+            htmlSize={14}
+            onChange={(e) => {
+              setError(undefined);
+              const num = Number(e.target.value);
+              if (isNaN(num)) {
+                setError("Må være et tall");
+              } else if (num > 2_147_483_647) {
```

Comment:
Hva er dette for et magisk tall `2_147_483_647`?
Kan det enten dras ut i en variabel som er mer forklarende, eller trengs den i det hele tatt?
Hva er det som bestemmer hva som er et for høyt beløp? Jeg vil feks. påstå at selv 2_147_483_646 er et for høyt beløp 😅

Reply:
Haha, var ikke helt sikker på hvorfor akkurat dette tallet ble validert på selv, så har ikke turt å røre den 🙈

Reply:
Ah, trodde du hadde lagt det inn. Ser det var der fra før. Har hørt i dev-kanalen om noen vedkjenner seg dette magiske tallet.